src/Searchable.php

```diff
@@ -26,38 +27,59 @@
  */
 class Searchable extends DataExtension
 {
-    public static $published_field = 'SS_Published';
+    public const TYPE_FIELD = 'type';
+
+    public const PUBLISHED_FIELD = 'SS_Published';
 
     /**
      * @config
      * @var array
      */
-    public static $mappings = array(
+    private static $elasticsearch_field_mappings = [
         'PrimaryKey'  => 'integer',
         'ForeignKey'  => 'integer',
-        'DBClassName' => 'string',
+        'DBClassName' => 'keyword',
         'DBDatetime'  => 'date',
         'Boolean'     => 'boolean',
         'Decimal'     => 'double',
         'Double'      => 'double',
-        'Enum'        => 'string',
+        'Enum'        => 'keyword',
         'Float'       => 'float',
-        'HTMLText'    => 'string',
-        'HTMLVarchar' => 'string',
+        'HTMLText'    => 'text',
+        'HTMLVarchar' => 'text',
         'Int'         => 'integer',
         'Datetime'    => 'date',
-        'Text'        => 'string',
-        'Varchar'     => 'string',
+        'Text'        => 'text',
+        'Varchar'     => 'text',
         'Year'        => 'integer',
         'File'        => 'attachment',
         'Date'        => 'date'
-    );
+    ];
+
+    /**
+     * ElasticSearch 7.0 compatibility: Use a custom 'type' field instead of deprecated _type
+     * @link https://www.elastic.co/guide/en/elasticsearch/reference/current/removal-of-types.html#_custom_type_field
```

Comment:
Will this just be ignored by older versions?

Reply:
`type` will work in older versions too